scripts/library.sh

```diff
@@ -475,21 +475,28 @@ function get_canonical_path() {
   echo "$(cd ${path%/*} && echo $PWD/${path##*/})"
 }
 
-# Return the base url we use to build the actual knative yaml sources.
-function get_knative_base_yaml_source() {
-  local knative_base_yaml_source="https://storage.googleapis.com/knative-nightly/@/latest"
+# Returns the URL to the latest manifest for the given Knative project.
+# Parameters: $1 - repository name of the given project
+#             $2 - name of the yaml file, without extension
+function get_latest_knative_yaml_source() {
   local branch_name=""
+  local repo_name="$1"
+  local yaml_name="$2"
   # Get the branch name from Prow's env var, see https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md.
   # Otherwise, try getting the current branch from git.
   (( IS_PROW )) && branch_name="${PULL_BASE_REF:-}"
   [[ -z "${branch_name}" ]] && branch_name="$(git rev-parse --abbrev-ref HEAD)"
-  # If it's a release branch, base URL should point to a specific version.
+  # If it's a release branch, the yaml source URL should point to a specific version.
   if [[ ${branch_name} =~ ^release-[0-9\.]+$ ]]; then
     # Get the latest tag name for the current branch, which is likely formatted as v0.5.0
     local tag_name="$(git describe --tags --abbrev=0)"
-    knative_base_yaml_source="https://storage.googleapis.com/knative-releases/@/previous/${tag_name}"
+    # The given repo might not have this tag, so we need to find its latest release manifest with the same major&minor version.
+    local major_minor="$(echo ${tag_name} | cut -d. -f-2)"
+    echo "$(gsutil ls gs://knative-releases/${repo_name}/previous/${major_minor}.*/${yaml_name}.yaml | tail -n 1)"
```

Comment:
You'll have to break this line into multiple steps because:
1. You're returning a gs URL, not an https URL.
1. gsutil ls doesn't guarantee the listing is sorted, so better use `sort` to ensure it.

Reply:
Sorry I forgot the gs URL when I made the change.
Done.